client/app/collections/search.coffee

```diff
@@ -0,0 +1,35 @@
+{Filtered}  = BackboneProjections
+
+
+module.exports = class Search extends Filtered
+    constructor: (underlying, options = {}) ->
+        app     = require 'application'
+        @search = app.model.get 'filter'
+
+        options.filter = @filter.bind @
+
+        super underlying, options
+
+        @listenTo app.model, 'change:filter', (model, value) ->
+            @search = value
+            @update()
+
+
+    filter: (model) ->
+        return true unless @search
+
+        filters = @search.match /(\w+):(\w+)/gi
+        filters.reduce (memo, filter) ->
```

Comment:
I think it will be more clear with [_.every](http://underscorejs.org/#every) from underscore/lodash.